httpclient5/src/main/java/org/apache/hc/client5/http/ssl/DefaultHostnameVerifier.java

```diff
@@ -355,7 +355,26 @@ static List<SubjectName> getSubjectAltNames(final X509Certificate cert, final in
                         if (o instanceof String) {
                             result.add(new SubjectName((String) o, type));
                         } else if (o instanceof byte[]) {
-                            // TODO ASN.1 DER encoded form
+                            final byte[] bytes = (byte[]) o;
+                            final String decodedValue;
+                            if (type == SubjectName.IP) {
+                                if (bytes.length == 4) {
+                                    decodedValue = byteArrayToIp(bytes); // IPv4
+                                } else if (bytes.length == 16) {
+                                    decodedValue = byteArrayToIPv6(bytes); // IPv6
+                                } else {
+                                    throw new IllegalArgumentException("Invalid byte length for IP address: " + bytes.length);
+                                }
+                            } else if (type == SubjectName.DNS) {
+                                throw new IllegalArgumentException("Unexpected byte[] for DNS SAN type");
+                            } else {
+                                if (LOG.isWarnEnabled()) {
+                                    LOG.warn("Unrecognized SAN type: {}, data: {}", type, TextUtils.toHexString(bytes));
+                                }
+                                decodedValue = TextUtils.toHexString(bytes); // Fallback to hex string
```

Comment:
@arturobernalg I would not do that. Before you know there will be so called security professionals with all sorts of hand-crafted certificates, claiming vulnerabilities in our code and demanding we issue CVE with them as a finder.

Reply:
@ok2c  please take another look